atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueReader.java

```diff
@@ -15,21 +15,40 @@
  */
 package com.palantir.atlasdb.sweep.queue;
 
-import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import java.util.function.IntSupplier;
 
 class SweepQueueReader {
     private final SweepableTimestamps sweepableTimestamps;
     private final SweepableCells sweepableCells;
+    private final IntSupplier maximumPartitionsToBatchInSingleRead;
```

Comment:
Ah didn't even realise you do in the end use an IntSupplier, so would definitely replace it by IntSupplier up the chain

Reply:
alright, will do. I really didn't want to leak the abstraction all the way here, though I thought e.g. in sweep-queue-factory it might be acceptable.